query_execution/WorkOrdersContainer.hpp

```diff
@@ -311,16 +311,17 @@ class WorkOrdersContainer {
     }
 
     inline void addWorkOrder(WorkOrder *workorder) {
-      workorders_.push(workorder);
+      workorders_.emplace(workorder);
```

Comment:
No @zuyu, that's not what the answer says. It says that ``emplace_back`` is not the correct choice when you are inserting a raw pointer that owns the resource, which is exactly our scenario. 

You can either construct the WorkOrder object (and thereby its pointer) right when you make the emplace call. That's not feasible in our case.

You need to call the ``make_unique`` function that's available in C++14. But considering that most of our code is C++11 compliant, I won't be a fan of using it here.